.gitignore

```diff
@@ -32,3 +33,6 @@
 # version files
 version.h
 
+# Visual Studio
+.vs/
+CMakeSettings.json
```

Comment:
Why the 3 additions here. I haven't seen an issue with Visual Studio leaving files in the source when I've used it but perhaps that's because I do my builds in `build` which is already ignored.

Reply:
These are relevant when using Visual Studio's native CMake integration. [You can work with CMake in VS without having to use the VS project generator](https://docs.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio?view=msvc-160). This means VS treats the project more like a solution, so it creates a `.vs` directory for its private files, and a `CMakeSettings.json` so you can configure CMake generation.
I added `/install*/` because that's a typical directory for the CMake install, and it's the default used by VS.